xds/internal/resolver: update github.com/cespare/xxhash to v2

tklauser opened the pull request:
github.com/cespare/xxhash/v2 supports Go ≥ 1.11 and this package states
1.11 in its go.mod file. The only symbol used from the xxhash package is the
Sum64String func which still exists and works the same in v2.

This gets rid of two indirect dependencies.

RELEASE NOTES: none

easwars commented:
Do we have an idea of what changed between v1.1.0 and v2.1.1? The xxhash repo doesn't include any release notes as part of their releases.

@zasweq : Any specific reason you chose v1.1.0 initially instead of v2.1.1 which has been out since on Nov 14, 2019?

zasweq commented:
Whoops, I didn't choose a specific version. This was chosen because it was already imported in Google3 third party.

tklauser commented:
> Do we have an idea of what changed between v1.1.0 and v2.1.1? The xxhash repo doesn't include any release notes as part of their releases.

I think the closest to something like release notes is this list of changes: https://github.com/cespare/xxhash/compare/v1.1.0...v2.1.1

From quickly skimming through them I don't see anything affecting current xxhash usage in grpc-go

zasweq reviewed (approved):
LGTM! Google3 already imports v2 of xxhash.

tklauser commented:
I've rebased the PR and added `RELEASE NOTES: none` to the PR description. It seems, the _Mergable_ check is still failing due to label and milestone not being set. Unfortunately, I cannot set these myself.

@zasweq Is there anything else I can do to move this forward?
